raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -379,16 +403,11 @@ public void transitionToFollower(
         int epoch,
         int leaderId
     ) {
+        int currentEpoch = state.epoch();
         if (localId.isPresent() && leaderId == localId.getAsInt()) {
             throw new IllegalStateException("Cannot transition to Follower with leaderId=" + leaderId +
                 " and epoch=" + epoch + " since it matches the local broker.id=" + localId);
-        } else if (!isVoter(leaderId)) {
-            throw new IllegalStateException("Cannot transition to Follower with leaderId=" + leaderId +
-                " and epoch=" + epoch + " since it is not one of the voters " + voters);
-        }
```

Comment:
In KIP-853, the leader may not be in the local set of voters. This can happen if there is an election right after a replica has been added to the set of voters and that state has not been replicated to all of the replicas.